doc/sphinx/source/sycl_with_pocl_arm.rst

```diff
@@ -0,0 +1,325 @@
+   ./opencl/CMakeLists.txt, at lines 23 and
+   24 <https://github.com/intel/llvm/blob/c5d04bcc0b7b7adf93a9f4c57faf6fada06575cf/opencl/CMakeLists.txt#L23>`__.
+
+POCL installation
+^^^^^^^^^^^^^^^^^
+
+1. After you have both the ICD-Loader and the Opencl Headers installed
+   you will need to set up the corresponding variables appropiately:
+
+::
+
+   VVV_ICD_LOADER=<YOUR ICD-LOADER PREFIX>
+   VVV_OCL_HEADERS=<YOUR OPENCL HEADERS PREFIX>
+   export CPLUS_INCLUDE_PATH=${VVV_OCL_HEADERS}/include:$CPLUS_INCLUDE_PATH/
+   export C_INCLUDE_PATH=${VVV_OCL_HEADERS}/include:$CPLUS_INCLUDE_PATH/
+   export PKG_CONFIG_PATH=${VVV_ICD_LOADER}/lib64/pkgconfig/:${VVV_OCL_HEADERS}/share/pkgconfig:$PKG_CONFIG_PATH
+   export LIBRARY_PATH=${VVV_ICD_LOADER}/lib64/:$LIBRARY_PATH
+   export LD_LIBRARY_PATH=${VVV_ICD_LOADER}/lib64:$LD_LIBRARY_PATH
+
+2. Clone the `POCL repository <https://github.com/pocl/pocl>`__, create
+   a build directory and from inside execute CMake. Remember to replace the paths accordingly:
+
+::
+
+   CC="clang" CXX="clang++" cmake -G Ninja -DCMAKE_INSTALL_PREFIX=<YOUR POCL INSTALLATION PREFIX> \
+   -DENABLE_ICD=ON -DCMAKE_PREFIX_PATH="<YOUR VANILLA LLVM PREFIX>;$VVV_ICD_LOADER;$VVV_OCL_HEADERS" \
+   -DENABLE_SPIR=ON -DENABLE_SPIRV=ON -DLLVM_SPIRV=<FULL PATH TO THE llvm-spirv BINARY FROM INTEL LLVM>  \
+   -DCMAKE_BUILD_TYPE=RelWithDebInfo -DSTATIC_LLVM=ON ..
+
+3. ninja install
+4. OPTIONAL: Run the test suite: ``ctest -j 128 -L internal``.
+5. When using POCL you should export some environment variables. I
+   suggest you create a env-pocl.sh script that exports the variables
+   for you:
+
+::
+
+   #!/bin/bash
+
+   BASE_PATH=<YOUR POCL INSTALLATION PREFIX>
+
+   # BIN
+   export PATH=${BASE_PATH}/bin:$PATH
+
+   # HEADERS
+   export CPLUS_INCLUDE_PATH=$CPLUS_INCLUDE_PATH:${BASE_PATH}/include/
+   export C_INCLUDE_PATH=$CPLUS_INCLUDE_PATH:${BASE_PATH}/include/
+
+   # LIBS
+   export LD_LIBRARY_PATH=${BASE_PATH}/lib64:$LD_LIBRARY_PATH
+   export LIBRARY_PATH=${BASE_PATH}/lib64:$LIBRARY_PATH
+
+   export PKG_CONFIG_PATH=${BASE_PATH}/lib64/pkgconfig/::$PKG_CONFIG_PATH
+
+   export OCL_ICD_VENDORS=${BASE_PATH}/etc/OpenCL/vendors/
+
+   # Variables for debugging programs
+   export VVV_pocl_help="SYCL_PI_TRACE=2 POCL_DEBUG=all OCL_ICD_ENABLE_TRACE=1"
+
+
+Using SYCL with POCL
+--------------------
+
+I uploaded a simple example to test if SYCL is working with POCL. It
+just tests that you can compile and execute simple SYCL programs using
+POCL as the OpenCL implementation that SYCL uses.
+
+1. ``source env-pocl.sh``
+2. ``source env-sycl.sh``
+3. The first test you should do is validate that the SYCL runtime can
+   find and query simple information from the POCL runtime. This is how
+   it looks for me:
+
+::
+
+   [host@user]$ sycl-ls
+   [opencl:cpu:0] Portable Computing Language, pthread-0xd01 OpenCL 3.0 PoCL HSTR: pthread-aarch64-unknown-linux-gnu-tsv110 [4.0-pre next-0-gbbb3d72]
```

Comment:
The PoCL version is different here which was listed above. Perhaps you can just more vaguely state it was tested with 4.0 as we are very close to branching?